apps/backend/src/authn/apikey.strategy.ts

```diff
@@ -1,23 +1,23 @@
 import {Injectable} from '@nestjs/common';
 import {PassportStrategy} from '@nestjs/passport';
 import HeaderAPIKeyStrategy from 'passport-headerapikey';
-import {ApiKeyService} from '../apikeys/apikey.service';
 import {User} from '../users/user.model';
+import {AuthnService} from './authn.service';
 
 @Injectable()
 export class APIKeyStrategy extends PassportStrategy(
   HeaderAPIKeyStrategy,
   'apikey'
 ) {
-  constructor(private readonly apiKeyService: ApiKeyService) {
+  constructor(private readonly authnService: AuthnService) {
     super(
       {header: 'Authorization', prefix: 'Api-Key '},
       false,
       async (
         apikey: string,
         done: (arg1: null, arg2: Promise<User | null> | boolean) => unknown
       ) => {
-        return done(null, this.apiKeyService.checkKey(apikey));
+        return done(null, this.authnService.validateApiKey(apikey));
```

Comment:
Still passing null to the exception parameter of `done`. What happens when you pass an exception based on the return value of `authnService.validateApiKey`?

Reply:
It continues the request without throwing an an error even if you provide an error to `done`.

Reply:
I just tested this and that is not the case

Reply:
Tried it again as well, it throws a 500 server error. Throwing exceptions within `validateApiKey` returns a more specific error

Reply:
That's now what I saw. It throws whatever error you pass into the first parameter. So if validateApiKey returns null, return create a new forbidden exception and pass it to the first parameter.